.golangci-kal.yml

```diff
@@ -0,0 +1,76 @@
+              - "notimestamp" # Prevents usage of 'Timestamp' fields
+              - "optionalfields" # Ensure that all fields marked as optional adhere to being pointers and
+                                 # having the `omitempty` value in their `json` tag where appropriate.
+              - "optionalorrequired" # Every field should be marked as `+optional` or `+required`.
+              - "requiredfields" # Required fields should not be pointers, and should not have `omitempty`.
+              - "ssatags" # Ensure array fields have the appropriate listType markers
+              - "statusoptional" # Ensure all first children within status should be optional.
+              - "statussubresource" # All root objects that have a `status` field should have a status subresource.
+              - "uniquemarkers" # Ensure that types and fields do not contain more than a single definition of a marker that should only be present once.
+
+            # Per discussion in July 2024, we are keeping phase fields for now.
+            # See https://github.com/kubernetes-sigs/cluster-api/pull/10897#discussion_r1685929508
+            # and https://github.com/kubernetes-sigs/cluster-api/pull/10897#discussion_r1685919394.
+            # - "nophase" # Phase fields are discouraged by the Kube API conventions, use conditions instead.
+
+            # Linters below this line are disabled, pending conversation on how and when to enable them.
+            disable:
+            - "*" # We will manually enable new linters after understanding the impact. Disable all by default.
+          lintersConfig:
+            conflictingmarkers:
+              conflicts:
+              - name: "default_vs_required"
+                sets:
+                  - ["default", "kubebuilder:default"]
+                  - ["required", "kubebuilder:validation:Required", "k8s:required"]
+                description: "A field with a default value cannot be required"
+            conditions:
+              isFirstField: Warn # Require conditions to be the first field in the status struct.
+              usePatchStrategy: Forbid # Forbid patchStrategy markers on the Conditions field.
+              useProtobuf: Forbid # We don't use protobuf, so protobuf tags are not required.
+            optionalfields:
+              pointers:
+                preference: WhenRequired # Always | WhenRequired # Whether to always require pointers, or only when required. Defaults to `Always`.
+                policy: SuggestFix # SuggestFix | Warn # The policy for pointers in optional fields. Defaults to `SuggestFix`.
+              omitempty:
+                policy: SuggestFix # SuggestFix | Warn | Ignore # The policy for omitempty in optional fields. Defaults to `SuggestFix`.
+
+  exclusions:
+    generated: strict
+    rules:
+    ## KAL should only run on API folders.
+    - path-except: "apis/kueue/*"
+      linters:
+        - kubeapilinter
```

Comment:
Why this?

Reply:
What happens if remove this? It looks like we are skipping other apis.

Reply:
We don't run the linter on the kueue api.

I can apply this to config and visibility also but I didn't see a reason there. They are not CRDs.

Reply:
Make sense.